### PR TITLE
Add Hybrid Quantum Neural Network (HQNN) tutorial

### PR DESCRIPTION
**Pull request by masa-whitestone**
## Summary
- Add HQNN tutorial (EN/JA) demonstrating end-to-end CNN + quantum variational circuit on Fashion-MNIST
- Uses Qamomile `@qkernel` API with `StatevectorEstimator` for exact expectation values
- Parameter shift rule implemented as custom `torch.autograd.Function` for quantum-classical gradient flow
- 2-conv ch=4,4 CNN architecture chosen to show meaningful quantum layer contribution (~+8% accuracy boost)

## Files
- `docs/en/vqa/hybrid_qnn.py` / `.ipynb` — English version
- `docs/ja/vqa/hybrid_qnn.py` / `.ipynb` — Japanese version
- `docs/{en,ja}/myst.yml` — Add HQNN to VQA section in TOC
- `docs/{en,ja}/index.md` — Add HQNN link to VQA section

## Test plan
- [ ] `uv run jupytext --to ipynb docs/en/vqa/hybrid_qnn.py` generates valid notebook
- [ ] `uv run jupyter nbconvert --execute --inplace docs/en/vqa/hybrid_qnn.ipynb` runs without error
- [ ] Same for JA version
- [ ] Doc build succeeds: `cd docs && make build-en && make build-ja`

### CODE REVIEW

File: docs/en/algorithm/hybrid_qnn.py [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+# # Hybrid Quantum Neural Network (HQNN)
+#
+# This tutorial demonstrates how to build a **Hybrid Quantum Neural Network** (HQNN) on Fashion-MNIST that combines classical neural network layers with a quantum variational circuit layer. The quantum layer is defined using Qamomile's `@qkernel` API, transpiled to Qiskit, and integrated into a PyTorch training pipeline via the parameter shift rule.
+#
+# ## Architecture
+#
+# The entire model — CNN feature extractor, quantum layer, and fusion classifier — is trained **end-to-end from scratch**:
+#
+# ```
+# Input Image → CNN → π·σ(·) → Quantum Layer → [CNN feats, Q output] → Classifier → Output
```
Comment (ksk-jij) [inline docs/en/algorithm/hybrid_qnn.py:30]: [question]
Could you add an explanation about π·σ(·)?

File: docs/en/vqa/hybrid_qnn.py [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+        for i in qmc.range(n_qubits):
+            q[i] = qmc.rz(q[i], weights[base + i])
+        for i in qmc.range(n_qubits):
+            q[i] = qmc.ry(q[i], weights[base + n_qubits + i])
+        for i in qmc.range(n_qubits):
+            q[i] = qmc.rz(q[i], weights[base + n_qubits * 2 + i])
+
+        # CNOT entanglement ladder
+        for i in qmc.range(n_qubits - 1):
+            q[i], q[i + 1] = qmc.cx(q[i], q[i + 1])
```
Comment (ksk-jij) [inline docs/en/vqa/hybrid_qnn.py]: [question]
Could we use `ry_layer`, `rz_layer` and`cx_entangling_layer` here instead of writing those naively?

File: docs/en/algorithm/hybrid_qnn.py [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+# %% [markdown]
+# Let's visualize the circuit structure:
+
+# %%
+variational_ansatz.draw(
+    n_qubits=N_QUBITS,
+    n_layers=N_LAYERS,
+    inputs=[0.1] * N_QUBITS,
+    weights=[0.1] * N_WEIGHTS,
+)
```
Comment (ksk-jij) [inline docs/en/algorithm/hybrid_qnn.py:113]: [my opinion]
If those inputs and weights do not have strong meanings, you can eliminate them. `draw` function still work without inputs and weights. Also, you can set `fold_loops=False` and `inline=False` if you could integrate `ry_layer`, `rz_layer` and `cx_entangling_layer` with your qkernel.

File: docs/en/vqa/hybrid_qnn.py [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+    variational_ansatz,
+    bindings={"n_qubits": N_QUBITS, "n_layers": N_LAYERS},
+    parameters=["inputs", "weights"],
+)
+
+# Get the underlying Qiskit circuit (with symbolic parameters)
+qiskit_circuit = executable.get_first_circuit()
+param_metadata = executable.compiled_quantum[0].parameter_metadata
+
+print(f"Circuit depth: {qiskit_circuit.depth()}")
```
Comment (ksk-jij) [inline docs/en/vqa/hybrid_qnn.py]: [my opinion]
Do you have any specific reasons why we print those? Like depth? If no, could we use that functionality? It can be used as follows.

```python
est = variational_ansatz.estimate_resources()
substituted_est = est.substitute(n_qubits=N_QUBITS, n_layers=N_LAYERS)
```

This does not give you depth, but gate counts. The number of parameters is directly the number of rotation gates.

Or, if you don't have strong meaning to see this, we may remove those prints completely.

File: docs/en/algorithm/hybrid_qnn.py
Comment (ksk-jij) [inline docs/en/algorithm/hybrid_qnn.py:1]: [my opinion]
In this notebook, mostly Qiskit is used; however, we have an executor, which can execute the circuit without writing qiskit. For instance, if you define `variational_ansatz` qkernel as

```python
@qmc.qkernel
def variational_ansatz(
    n_qubits: qmc.UInt,
    n_layers: qmc.UInt,
    inputs: qmc.Vector[qmc.Float],
    weights: qmc.Vector[qmc.Float],
    hamiltonian: qmc.Observable
) -> qmc.Float:
```

and define the observables for <Z_i> as

```python
observables = []
for i in range(N_QUBITS):
    observable = qmo.Hamiltonian(num_qubits=N_QUBITS)
    observable += qmo.Z(i)
    observables.append(observable)
```

Then, you can create a sequence of executable objects as

```python
executables = [
    transpiler.transpile(
        variational_ansatz,
        bindings={"n_qubits": N_QUBITS, "n_layers": N_LAYERS, "hamiltonian": hamiltonian},
        parameters=["inputs", "weights"],
    )
    for hamiltonian in observables
]
```

In this case, I think we can re-define `quantum_forward` function such as 

```python
def quantum_forward(input_vals: np.ndarray, weight_vals: np.ndarray) -> np.ndarray:
    runtime_bindings = {
        "inputs": list(input_vals),
        "weights": list(weight_vals),
    }
    return np.array(
        [exe.run(transpiler.executor(), bindings=runtime_bindings).result() for exe in executables],
        dtype=float,
    )
```

I wonder what you think of this idea.

File: docs/en/vqa/hybrid_qnn.py [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+
+            # Gradient w.r.t. weights
+            for k in range(len(weights_np)):
+                shifted_plus = weights_np.copy()
+                shifted_plus[k] += SHIFT
+                shifted_minus = weights_np.copy()
+                shifted_minus[k] -= SHIFT
+
+                fwd_plus = quantum_forward(inp_np, shifted_plus)
+                fwd_minus = quantum_forward(inp_np, shifted_minus)
```
Comment (ksk-jij) [inline docs/en/vqa/hybrid_qnn.py]: [my opinion]
This feels a bit lux in terms of memory usage since every time we copy the weights. What do you think of the idea of writing this as

```python
weights_np[k] += SHIFT
fwd_plus = quantum_forward(inp_np, weights_np)
weights_np[k] -= SHIFT

weights_np[k] -= SHIFT
fwd_minus = quantum_forward(inp_np, shifted_minus)
weights_np[k] += SHIFT
```

File: docs/en/vqa/hybrid_qnn.py [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+
+            # Gradient w.r.t. inputs
+            for k in range(len(inp_np)):
+                shifted_plus = inp_np.copy()
+                shifted_plus[k] += SHIFT
+                shifted_minus = inp_np.copy()
+                shifted_minus[k] -= SHIFT
+
+                fwd_plus = quantum_forward(shifted_plus, weights_np)
+                fwd_minus = quantum_forward(shifted_minus, weights_np)
```
Comment (ksk-jij) [inline docs/en/vqa/hybrid_qnn.py]: [my opinion]
This feels a bit lux in terms of memory usage since every time we copy the inp_np. What do you think of the same idea as weights comment.

File: docs/en/vqa/hybrid_qnn.ipynb
Comment (ksk-jij) [inline docs/en/vqa/hybrid_qnn.ipynb:1]: [question]
Could you remove the following error at the last cell? This is probably due to the settings of execution. The current `main` should have been addressed this, so if you haven't merged the current main, please merge it.

```text
/var/folders/x9/z9h0851n643d8wc3j2v8frv40000gn/T/ipykernel_40664/2214683661.py:53: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  plt.show()
```

**Review by ksk-jij — APPROVED** [review]
@masa-whitestone Thank you for the PR! I've left comments! Please have a look.

**Review by ksk-jij — COMMENTED** [review]
Also, could you skip this docs in the test?

File: docs/en/vqa/hybrid_qnn.py (hunk 2 above)
Comment (ksk-jij) [reply docs/en/vqa/hybrid_qnn.py]: @masa-whitestone You are now using `cz_entangling_layer` instead of `cx_entangling_layer`. Is is intentional?

**Review by ksk-jij — APPROVED** [review]
@masa-whitestone Thank you for your all efforts! Looks good to me :)
